test/es-module/test-http-imports.mjs

```diff
@@ -112,6 +112,21 @@ for (const { protocol, createServer } of [
     assert.strict.notEqual(redirectedNS.default, ns.default);
     assert.strict.equal(redirectedNS.url, url.href);
 
+    // Redirects have the same import.meta.url but different cache
+    // entry on Web
+    const relativeAfterRedirect = new URL(url.href + 'foo/index.js');
+    const redirected = new URL(url.href + 'bar/index.js');
```

Comment:
nit: Doesn't the `URL` constructor already handle this?

```suggestion
    const relativeAfterRedirect = new URL( 'foo/index.js', url.href);
    const redirected = new URL('bar/index.js', url.href);
```

Reply:
🤷🏼 no preference